cli/cmd/uninstall.go

```diff
@@ -0,0 +1,304 @@
+	if err != nil {
+		return "", err
+	}
+	return m, nil
+}
+
+// Although role bindings are namespaced resources in nature
+// some admin role bindings are created and persisted in the kube-system namespace and will not be deleted
+// when the namespace is deleted
+func fetchKubeSystemRoleBindings(k *k8s.KubernetesAPI, options metav1.ListOptions) (string, error) {
+	list, err := k.RbacV1().RoleBindings("kube-system").List(options)
+	if err != nil {
+		return "", err
+	}
+
+	if list.Items == nil {
+		return "", nil
+	}
+
+	m, err := marshalKubernetesResource(list)
+	if err != nil {
+		return "", err
+	}
+	return m, nil
+}
+
+func fetchNamespaceResource(k *k8s.KubernetesAPI) (string, error) {
+	obj, err := k.CoreV1().Namespaces().Get(controlPlaneNamespace, metav1.GetOptions{})
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return "", nil
+		}
+		return "", err
+	}
+
+	m, err := marshalKubernetesResource(obj)
+	if err != nil {
+		return "", err
+	}
+	return m, nil
+}
+
+func fetchPodSecurityPolicy(k *k8s.KubernetesAPI, options metav1.ListOptions) (string, error) {
+	list, err := k.PolicyV1beta1().PodSecurityPolicies().List(options)
+	if err != nil {
+		return "", err
+	}
+
+	if list.Items == nil {
+		return "", nil
+	}
+
+	m, err := marshalKubernetesResource(list)
+	if err != nil {
+		return "", err
+	}
+	return m, nil
+}
+
+func fetchValidatingWebhooksConfiguration(k *k8s.KubernetesAPI, options metav1.ListOptions) (string, error) {
+	list, err := k.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().List(options)
+	if err != nil {
+		return "", err
+	}
+
+	if list.Items == nil {
+		return "", nil
+	}
+
+	m, err := marshalKubernetesResource(list)
+	if err != nil {
+		return "", err
+	}
+	return m, nil
+}
+
+func fetchMutatingWebhooksConfiguration(k *k8s.KubernetesAPI, options metav1.ListOptions) (string, error) {
+	list, err := k.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().List(options)
+	if err != nil {
+		return "", err
+	}
+
+	if list.Items == nil {
+		return "", nil
+	}
+
+	m, err := marshalKubernetesResource(list)
+	if err != nil {
+		return "", err
+	}
+	return m, nil
+}
+func fetchAPIRegistrationResources(k *k8s.KubernetesAPI, options metav1.ListOptions) (string, error) {
+	apiClient, err := apiregistrationv1client.NewForConfig(k.Config)
+	if err != nil {
+		return "", nil
+	}
+
+	list, err := apiClient.APIServices().List(options)
+	if err != nil {
+		return "", nil
+	}
+
+	if list.Items == nil {
+		return "", nil
+	}
+
+	m, err := marshalKubernetesResource(list)
+	if err != nil {
+		return "", err
+	}
+	return m, nil
+}
+
+func marshalKubernetesResource(resource runtime.Object) (string, error) {
+	var gvk = resource.GetObjectKind().GroupVersionKind()
+	if gvk.Empty() {
+		gvks, _, err := k8s.ObjectKinds(resource)
+		if err == nil && len(gvks) > 0 {
+			gvk = gvks[0]
+		}
+		resource.GetObjectKind().SetGroupVersionKind(gvk)
+	}
+	b, err := yaml.Marshal(resource)
```

Comment:
Technically, we only need to marshal the name, namespace (if applicable), kind, and apiVersion, right?  I think we can maybe get away with just marshalling an ObjectReference instead of the entire resource.

Reply:
@adleong yes, completely right! It was an oversight on my part, was under the impression it did not work just with the kind, name and apiVersion. Will make changes.